packages/fxa-content-server/app/scripts/lib/channels/receivers/web-channel.js

```diff
@@ -31,8 +31,12 @@ _.extend(WebChannelReceiver.prototype, Backbone.Events, {
   },
 
   receiveMessage(event) {
-    const detail = event.detail;
-
+    let detail;
+    try {
+      detail = JSON.parse(event.detail);
+    } catch (e) {
+      detail = event.detail;
+    }
```

Comment:
I'm curious why this change was necessary, is it a limitation of the webextension webchannel receiver?

I'd also be tempted to do a `typeof event.details` here to ensure you don't try to `JSON.parse` something that's not a string.

Reply:
WebExtensions in Geckoview/ Fenix cannot pass js objects, only strings. Otherwise it displays `[protected]`, `[inaccessible]` or something

Reply:
Hrm, I also recall an ancient bug about sending only strings on Desktop as well, which has had to get added to a special allowlist in order to bypass.

Reply:
It's when *sending* an event to the browser that detail has to be stringified: 

* https://github.com/mozilla/fxa/blob/master/packages/fxa-content-server/app/scripts/lib/channels/senders/web-channel.js#L117
* https://bugzilla.mozilla.org/show_bug.cgi?id=1275616
* https://bugzilla.mozilla.org/show_bug.cgi?id=1238128